src/storage/src/hummock/store/version.rs

```diff
@@ -729,119 +729,86 @@ impl HummockVersionReader {
         let mut non_overlapping_iters = Vec::new();
         let mut overlapping_iters = Vec::new();
         let mut overlapping_iter_count = 0;
-        let mut fetch_meta_reqs = vec![];
+        let timer = self
+            .state_store_metrics
+            .iter_fetch_meta_duration
+            .with_label_values(&[table_id_label])
+            .start_timer();
+
+        let mut sst_read_options = SstableIteratorReadOptions::from_read_options(&read_options);
+        if read_options.prefetch_options.exhaust_iter {
+            sst_read_options.must_iterated_end_user_key =
+                Some(user_key_range.1.map(|key| key.cloned()));
+        }
+        let sst_read_options = Arc::new(sst_read_options);
         for level in committed.levels(read_options.table_id) {
             if level.table_infos.is_empty() {
                 continue;
             }
 
             if level.level_type == LevelType::Nonoverlapping as i32 {
                 let table_infos = prune_nonoverlapping_ssts(&level.table_infos, user_key_range_ref);
-
-                let fetch_meta_req = table_infos
+                let sstables = table_infos
                     .filter(|sstable_info| {
                         sstable_info
                             .table_ids
                             .binary_search(&read_options.table_id.table_id)
                             .is_ok()
                     })
+                    .cloned()
                     .collect_vec();
-                fetch_meta_reqs.push((level.level_type, fetch_meta_req));
-            } else {
-                let table_infos = prune_overlapping_ssts(
-                    &level.table_infos,
-                    read_options.table_id,
-                    &table_key_range,
-                );
-                // Overlapping
-                let fetch_meta_req = table_infos.rev().collect_vec();
-                if !fetch_meta_req.is_empty() {
-                    fetch_meta_reqs.push((level.level_type, fetch_meta_req));
+                if sstables.is_empty() {
+                    continue;
                 }
-            }
-        }
-        let mut flatten_reqs = vec![];
-        let mut req_count = 0;
-        for (_, fetch_meta_req) in &fetch_meta_reqs {
-            for sstable_info in fetch_meta_req {
-                let inner_req_count = req_count;
-                let capture_ref = async {
-                    // We would fill block to high priority cache for level-0
-                    self.sstable_store
-                        .sstable_syncable(sstable_info, &local_stats)
+                if sstables.len() > 1 {
```

Comment:
IMO， we need some doc to explain that we remove `hit_sstable_bloom_filter` when non_overlapping sstable.size() > 1. It may result in better code readability

Reply:
> IMO， we need some doc to explain that we remove `hit_sstable_bloom_filter` when non_overlapping sstable.size() > 1. It may result in better code readability

+1